spec/requests/editor_spec.rb

```diff
@@ -2,27 +2,35 @@
 
 RSpec.describe "Editor", type: :request do
   describe "GET /new" do
-    context "when not logged-in" do
-      it "asks the non logged in user to sign in" do
-        get new_path
+    subject(:request_call) { get new_path }
 
-        # NOTE: [@jeremyf] This response status seems a bit surprising.  Should the prompt for signin redirect
-        # the user to a login page?  Or are we doing something with Javascript?  For now, paint me
-        # curious, but not enough to pull on this thread.
-        expect(response).to have_http_status(:ok)
-      end
+    let(:user) { create(:user) }
+
+    context "when not authenticated" do
+      it { within_block_is_expected.to raise_error ApplicationPolicy::UserRequiredError }
     end
 
-    context "when email login is allowed in /admin/customization/config" do
+    context "when authenticated but not authorized" do
       before do
-        allow(Settings::Authentication).to receive(:allow_email_password_login).and_return(true)
+        login_as user
+        allow(ArticlePolicy).to receive(:limit_post_creation_to_admins?).and_return(true)
       end
 
-      it "asks the non logged in user to sign in, with email signin enabled" do
-        get new_path
+      # [@jeremyf] We're handling the authentication and authorization exceptions just a bit
+      #            differently.  In this case (e.g. they don't have permission) we are relying on
+      #            the application configuration to gracefully handle the authorization error (as it
+      #            has prior and up to <2022-02-17 Thu>).
+      it { within_block_is_expected.to raise_error(Pundit::NotAuthorizedError) }
+    end
+
+    context "when authenticated and authorized" do
+      before { login_as user }
 
-        expect(response.body).to include("Email")
-        expect(response.body).to include("Password")
+      it "is a successful response" do
+        # We have lots of Cypress tests of the behavior of the `/new` page.  Let's make sure we're
+        # verifying AuthN/AuthZ things.
+        get new_path
+        expect(response).to be_successful
```

Comment:
IIRC, the `be_successful` RSpec matcher [invokes the `successful?` predicate method](https://relishapp.com/rspec/rspec-expectations/v/3-10/docs/built-in-matchers/predicate-matchers), which will return an opaque error message like `expected successful? to return true, got false`, so you don't know what the actual HTTP status code was. The `have_http_status` matcher provided by `rspec-rails` tells you what the actual status was.

```suggestion
        expect(response).to have_http_status(:ok)
```